propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagators/ot_trace/__init__.py

```diff
@@ -0,0 +1,177 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from re import compile as re_compile
+from typing import Iterable, Optional
+
+from opentelemetry.baggage import get_all, set_baggage
+from opentelemetry.context import Context
+from opentelemetry.trace import (
+    INVALID_SPAN_ID,
+    INVALID_TRACE_ID,
+    DefaultSpan,
+    SpanContext,
+    TraceFlags,
+    get_current_span,
+    set_span_in_context,
+)
+from opentelemetry.trace.propagation.textmap import (
+    Getter,
+    Setter,
+    TextMapPropagator,
+    TextMapPropagatorT,
+)
+
+OT_TRACE_ID_HEADER = "ot-tracer-traceid"
+OT_SPAN_ID_HEADER = "ot-tracer-spanid"
+OT_SAMPLED_HEADER = "ot-tracer-sampled"
+OT_BAGGAGE_PREFIX = "ot-baggage-"
+
+_valid_header_name = re_compile(r"^[\w_^`!#$%&'*+.|~]+$")
+_valid_header_value = re_compile(r"^[\t\x20-\x7e\x80-\xff]+$")
+_valid_extract_traceid = re_compile(r"[0-9a-f]{32}|[0-9a-f]{16}")
```

Comment:
I'm a little confused about this. Will the incoming `ot-tracer-traceid` header always be 64 bit/16 hex places or is it unpadded? Will it ever be more than 16 hex places (we allow 32 here)? Looking quickly at [the opentracing impl](https://github.com/opentracing/basictracer-python/blob/0688470e034f6f5c89c2cc60181822d9982048db/basictracer/text_propagator.py#L50-L51) you linked, it doesn't seem to pad out the hex to full 16 hex places. The injector in this PR does not pad out the full 16 bits either (e.g. it would convert trace id `123` into `7b`), which would fail this regex.

Reply:
All right, fixed the checks